Can't find "cholmod.h" while installing using pip on macOS

I am installing scikit-sparse via pip and I got the error below:
```
 Building wheel for scikit-sparse (PEP 517) ... error
  ERROR: Command errored out with exit status 1:
   command: /opt/local/Library/Frameworks/Python.framework/Versions/3.8/bin/python3.8 /opt/local/Library/Frameworks/Python.framework/Versions/3.8/lib/python3.8/site-packages/pip/_vendor/pep517/in_process/_in_process.py build_wheel /var/folders/zl/498rsyv11_v53_k5x1xfflkw0000gn/T/tmphqddsg3a
       cwd: /private/var/folders/zl/498rsyv11_v53_k5x1xfflkw0000gn/T/pip-install-0ezcix_f/scikit-sparse_63d2a4408dfe4ad1a24ef3837a8eea73
  Complete output (35 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-11.3-x86_64-3.8
  creating build/lib.macosx-11.3-x86_64-3.8/sksparse
  copying sksparse/__init__.py -> build/lib.macosx-11.3-x86_64-3.8/sksparse
  copying sksparse/test_cholmod.py -> build/lib.macosx-11.3-x86_64-3.8/sksparse
  creating build/lib.macosx-11.3-x86_64-3.8/sksparse/test_data
  copying sksparse/test_data/illc1033.mtx.gz -> build/lib.macosx-11.3-x86_64-3.8/sksparse/test_data
  copying sksparse/test_data/illc1033_rhs1.mtx.gz -> build/lib.macosx-11.3-x86_64-3.8/sksparse/test_data
  copying sksparse/test_data/well1033_rhs1.mtx.gz -> build/lib.macosx-11.3-x86_64-3.8/sksparse/test_data
  copying sksparse/test_data/well1033.mtx.gz -> build/lib.macosx-11.3-x86_64-3.8/sksparse/test_data
  copying sksparse/test_data/well1850_rhs1.mtx.gz -> build/lib.macosx-11.3-x86_64-3.8/sksparse/test_data
  copying sksparse/test_data/illc1850.mtx.gz -> build/lib.macosx-11.3-x86_64-3.8/sksparse/test_data
  copying sksparse/test_data/illc1850_rhs1.mtx.gz -> build/lib.macosx-11.3-x86_64-3.8/sksparse/test_data
  copying sksparse/test_data/well1850.mtx.gz -> build/lib.macosx-11.3-x86_64-3.8/sksparse/test_data
  running build_ext
  building 'sksparse.cholmod' extension
  creating build/temp.macosx-11.3-x86_64-3.8
  creating build/temp.macosx-11.3-x86_64-3.8/sksparse
  /usr/bin/clang -Wno-unused-result -Wsign-compare -Wunreachable-code -fno-common -dynamic -DNDEBUG -g -fwrapv -O3 -Wall -pipe -Os -isysroot/Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk -Isksparse -I/private/var/folders/zl/498rsyv11_v53_k5x1xfflkw0000gn/T/pip-build-env-o0pnraoh/overlay/lib/python3.8/site-packages/numpy/core/include -I/opt/local/Library/Frameworks/Python.framework/Versions/3.8/include -I/usr/include/suitesparse -I/opt/local/Library/Frameworks/Python.framework/Versions/3.8/include/python3.8 -c sksparse/cholmod.c -o build/temp.macosx-11.3-x86_64-3.8/sksparse/cholmod.o
  In file included from sksparse/cholmod.c:638:
  In file included from /private/var/folders/zl/498rsyv11_v53_k5x1xfflkw0000gn/T/pip-build-env-o0pnraoh/overlay/lib/python3.8/site-packages/numpy/core/include/numpy/arrayobject.h:4:
  In file included from /private/var/folders/zl/498rsyv11_v53_k5x1xfflkw0000gn/T/pip-build-env-o0pnraoh/overlay/lib/python3.8/site-packages/numpy/core/include/numpy/ndarrayobject.h:12:
  In file included from /private/var/folders/zl/498rsyv11_v53_k5x1xfflkw0000gn/T/pip-build-env-o0pnraoh/overlay/lib/python3.8/site-packages/numpy/core/include/numpy/ndarraytypes.h:1830:
  /private/var/folders/zl/498rsyv11_v53_k5x1xfflkw0000gn/T/pip-build-env-o0pnraoh/overlay/lib/python3.8/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:17:2: warning: "Using deprecated NumPy API, disable it with "          "#define NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-W#warnings]
  #warning "Using deprecated NumPy API, disable it with " \
   ^
  In file included from sksparse/cholmod.c:640:
  sksparse/cholmod_backward_compatible.h:1:10: fatal error: 'cholmod.h' file not found
  #include "cholmod.h"
           ^~~~~~~~~~~
  1 warning and 1 error generated.
  error: command '/usr/bin/clang' failed with exit status 1
  ----------------------------------------
 ERROR: Failed building wheel for scikit-sparse
Failed to build scikit-sparse
ERROR: Could not build wheels for scikit-sparse which use PEP 517 and cannot be installed directly
```
And I already installed suitesparse via MacPorts, it seems it is trying to find "cholmod.h" in `/usr/include` which does not even exist, and the suitesparse files are in `/opt/local/include`. Is there a way to modify the include path for pip?
![image](https://user-images.githubusercontent.com/15608306/120813832-dd0c7900-c580-11eb-9dd9-3b82fa21cf3b.png)

--- a/.gitignore
+++ b/.gitignore
@@ -86,3 +86,7 @@ Thumbs.db
 
 # virtual environments
 venv
+
+# vscode
+.devcontainer/
+.vscode/

--- a/README.md
+++ b/README.md
@@ -1,7 +1,7 @@
 [![GitHub release (latest by date)](https://img.shields.io/github/v/release/scikit-sparse/scikit-sparse)](https://github.com/scikit-sparse/scikit-sparse/releases/latest)
 [![PyPI](https://img.shields.io/pypi/v/scikit-sparse)](https://pypi.org/project/scikit-sparse/)
 [![Conda Version](https://img.shields.io/conda/vn/conda-forge/scikit-sparse.svg)](https://anaconda.org/conda-forge/scikit-sparse)
-[![GitHub Workflow Status (event)](https://img.shields.io/github/workflow/status/scikit-sparse/scikit-sparse/CI%20targets?label=CI%20Tests)](https://github.com/scikit-sparse/scikit-sparse/actions/workflows/ci_test.yml) 
+[![GitHub Workflow Status (event)](https://img.shields.io/github/workflow/status/scikit-sparse/scikit-sparse/CI%20targets?label=CI%20Tests)](https://github.com/scikit-sparse/scikit-sparse/actions/workflows/ci_test.yml)
 [![Python Versions](https://img.shields.io/badge/python-3.6%2C%203.7%2C%203.8%2C%203.9-blue.svg)]()
 [![GitHub license](https://img.shields.io/github/license/scikit-sparse/scikit-sparse)](https://github.com/scikit-sparse/scikit-sparse/blob/master/LICENSE.txt)
 
@@ -30,6 +30,16 @@ sudo apt-get install libsuitesparse-dev
 Then, `scikit-sparse` can be installed via pip:
 ```bash
 pip install scikit-sparse
+```
+
+If you suite-sparse library is installed in a non-standard place and you get errors when installing with `pip` you can use the environment
+variables:
+* `SUITESPARSE_INCLUDE_DIR`
+* `SUITESPARSE_LIBRARY_DIR`
+
+at runtime so the compiler can find them. For example, lets say your suite-sparse installation is in `/opt/local` then you can run
+```bash
+SUITESPARSE_INCLUDE_DIR=/opt/local/include SUITESPARSE_LIBRARY_DIR=/opt/local/lib pip install scikit-sparse
 ```
 
 ### With `conda`

--- a/setup.py
+++ b/setup.py
@@ -10,7 +10,7 @@
 # 2016-2017   Joscha Reimer           <jor@informatik.uni-kiel.de>
 # 2021-       Justin Ellis            <justin.ellis18@gmail.com>
 
-"""Sparse matrix tools. 
+"""Sparse matrix tools.
 
 This is a home for sparse matrix code in Python that plays well with
 scipy.sparse, but that is somehow unsuitable for inclusion in scipy
@@ -20,6 +20,13 @@ So far we have a wrapper for the CHOLMOD library for sparse Cholesky
 decomposition. Further contributions are welcome!
 """
 
+import os
+import sys
+
+import numpy as np
+from Cython.Build import cythonize
+from setuptools import Extension, find_packages, setup
+
 DISTNAME = "scikit-sparse"
 DESCRIPTION = "Scikit sparse matrix package"
 LONG_DESCRIPTION = __doc__
@@ -28,11 +35,23 @@ MAINTAINER_EMAIL = "justin.ellis18@gmail.com"
 URL = "https://github.com/scikit-sparse/scikit-sparse"
 LICENSE = "BSD"
 
-import sys
 
-import numpy as np
-from Cython.Build import cythonize
-from setuptools import Extension, find_packages, setup
+INCLUDE_DIRS = [
+    np.get_include(),
+    sys.prefix + "/include",
+    # Debian's suitesparse-dev installs to
+    # /usr/include/suitesparse
+    "/usr/include/suitesparse",
+]
+LIBRARY_DIRS = []
+
+user_include_dir = os.getenv("SUITESPARSE_INCLUDE_DIR")
+user_library_dir = os.getenv("SUITESPARSE_LIBRARY_DIR")
+if user_include_dir:
+    INCLUDE_DIRS.append(user_include_dir)
+
+if user_library_dir:
+    LIBRARY_DIRS.append(user_library_dir)
 
 setup(
     install_requires=["numpy>=1.13.3", "scipy>=0.19"],
@@ -70,14 +89,8 @@ setup(
         Extension(
             "sksparse.cholmod",
             ["sksparse/cholmod.pyx"],
-            include_dirs=[
-                np.get_include(),
-                sys.prefix + "/include",
-                # Debian's suitesparse-dev installs to
-                # /usr/include/suitesparse
-                "/usr/include/suitesparse",
-            ],
-            library_dirs=[],
+            include_dirs=INCLUDE_DIRS,
+            library_dirs=LIBRARY_DIRS,
             libraries=["cholmod"],
         )
     ),

